File: jdroid-java-core/src/main/java/com/jdroid/java/utils/ObjectUtils.java
package com.jdroid.java.utils;

/**
 * <p>
 * Operations on <code>Object</code>.
 * </p>
 */
public class ObjectUtils {
	
	/**
	 * <p>
	 * Compares two objects for equality, where either one or both objects may be <code>null</code>.
	 * </p>
	 * 
	 * <pre>
	 * ObjectUtils.equals(null, null)                  = true
	 * ObjectUtils.equals(null, "")                    = false
	 * ObjectUtils.equals("", null)                    = false
	 * ObjectUtils.equals("", "")                      = true
	 * ObjectUtils.equals(Boolean.TRUE, null)          = false
	 * ObjectUtils.equals(Boolean.TRUE, "true")        = false
	 * ObjectUtils.equals(Boolean.TRUE, Boolean.TRUE)  = true
	 * ObjectUtils.equals(Boolean.TRUE, Boolean.FALSE) = false
	 * </pre>
	 * 
	 * @param object1 the first object, may be <code>null</code>
	 * @param object2 the second object, may be <code>null</code>
	 * @return <code>true</code> if the values of both objects are the same
	 */
	public static boolean equals(Object object1, Object object2) {
		if (object1 == object2) {
			return true;
		}
		if ((object1 == null) || (object2 == null)) {
			return false;
		}
		return object1.equals(object2);
	}
	
}

File: jdroid-java-core/src/main/java/com/jdroid/java/utils/ObjectUtils.kt
package com.jdroid.java.utils

@Deprecated(message = "")
object ObjectUtils {

    /**
     *
     *
     * Compares two objects for equality, where either one or both objects may be `null`.
     *
     *
     *
     * <pre>
     * ObjectUtils.equals(null, null)                  = true
     * ObjectUtils.equals(null, "")                    = false
     * ObjectUtils.equals("", null)                    = false
     * ObjectUtils.equals("", "")                      = true
     * ObjectUtils.equals(Boolean.TRUE, null)          = false
     * ObjectUtils.equals(Boolean.TRUE, "true")        = false
     * ObjectUtils.equals(Boolean.TRUE, Boolean.TRUE)  = true
     * ObjectUtils.equals(Boolean.TRUE, Boolean.FALSE) = false
    </pre> *
     *
     * @param object1 the first object, may be `null`
     * @param object2 the second object, may be `null`
     * @return `true` if the values of both objects are the same
     */
    fun equals(object1: Any?, object2: Any?): Boolean {
        if (object1 === object2) {
            return true
        }
        return if (object1 == null || object2 == null) {
            false
        } else object1 == object2
    }
}
